File: zantt-api-backend/src/main/java/com/zantt/services/TaskService.java
package com.zantt.services;

import com.zantt.entities.TaskEntity;
import com.zantt.exceptions.WellKnownApiException;
import com.zantt.repositories.ProjectRepository;
import com.zantt.repositories.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public class TaskService {
    @Autowired
    TaskRepository taskRepository;
    @Autowired
    ProjectRepository projectRepository;

    public List<TaskEntity> getTasksByProjectId(String projectId) {
        var project = projectRepository.findById(projectId);
        if (project.isEmpty()) {
            throw new WellKnownApiException("Can not found project", "PROJECT_NOT_FOUND");
        }

        return taskRepository.findByProjectId(projectId);
    }

    public TaskEntity addTask(String projectId, String title) {
        var project = projectRepository.findById(projectId);
        if (project.isEmpty()) {
            throw new WellKnownApiException("Can not found project", "PROJECT_NOT_FOUND");
        }

        var taskId = UUID.randomUUID().toString();

        var task = new TaskEntity();
        task.setTaskId(taskId);
        task.setProjectId(projectId);
        task.setTitle(title);

        return taskRepository.save(task);
    }

    public void deleteTask(String taskId) {
        taskRepository.deleteById(taskId);
    }
}

File: zantt-api-backend-net/Zantt/Services/TaskService.cs
using Zantt.Entities;
using Zantt.Exceptions;
using Zantt.Repositories;

namespace Zantt.Services;

public class TaskService
{
    private readonly TaskRepository taskRepository;
    private readonly ILogger<TaskService> logger;

    public TaskService(
        ILogger<TaskService> logger,
        TaskRepository taskRepository)
    {
        this.logger = logger;
        this.taskRepository = taskRepository;
    }

    public List<TaskEntity> GetTasks(string projectId)
    {
        return taskRepository.GetTasksByProjectId(projectId);
    }

    public TaskEntity? GetTask(string taskId)
    {
        return taskRepository.GetTaskByTaskId(taskId);
    }

    public TaskEntity? AddTask(string projectId, string title)
    {
        if (projectId == null)
        {
            throw new WellKnownApiException($"{nameof(projectId)} is null", "INVALID_PARAMETER");
        }
        if (title == null)
        {
            throw new WellKnownApiException($"{nameof(title)} is null", "INVALID_PARAMETER");
        }

        return taskRepository.AddTask(projectId, title);
    }

    public TaskEntity? UpdateTask(string taskId, string title)
    {
        if (taskId == null)
        {
            throw new WellKnownApiException($"{nameof(taskId)} is null", "INVALID_PARAMETER");
        }
        if (title == null)
        {
            throw new WellKnownApiException($"{nameof(title)} is null", "INVALID_PARAMETER");
        }

        return taskRepository.UpdateTaskByTaskId(taskId, title);
    }

    public void DeleteTask(string taskId)
    {
        taskRepository.DeleteTaskByTaskId(taskId);
    }
}
